picky/src/ssh/mod.rs

```diff
@@ -8,13 +8,11 @@ pub use certificate::{SshCertKeyType, SshCertType, SshCertificate, SshCertificat
 pub use private_key::SshPrivateKey;
 pub use public_key::SshPublicKey;
 
-use base64::read::DecoderReader;
-use base64::write::EncoderWriter;
 use byteorder::ReadBytesExt;
 use std::io::{self, Read};
 
-pub type Base64Writer<T> = EncoderWriter<T>;
-pub type Base64Reader<'a, R> = DecoderReader<'a, R>;
+pub type Base64Writer<'a, T, E> = base64::write::EncoderWriter<'a, T, E>;
+pub type Base64Reader<'a, T, E> = base64::read::DecoderReader<'a, T, E>;
```

Comment:
I know this wasn’t originally your code, but do you think you can make these public at crate-level only?
I feel this should not be exposed as part of the public API of picky

```suggestion
pub(crate) type Base64Writer<'a, T, E> = base64::write::EncoderWriter<'a, T, E>;
pub(crate) type Base64Reader<'a, T, E> = base64::read::DecoderReader<'a, T, E>;
```

Reply:
Sure, I'll fix this